### PR TITLE
media: Add state logging macro

### PR DESCRIPTION
**Pull request by junmin-kim**
LOG_STATE() print current state

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3bff47d3e347725280418fc214389db97fc15b3f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3bff47d3e347725280418fc214389db97fc15b3f] - Code Rule Check OK.

**Comment by junmin-kim** [comment]
Do not merge

**Comment by seinfra** [comment]
Target : [f4cf5dd56e4cd91e615e0fef4e1452597df50a8b] -

**Comment by seinfra** [comment]
Target : [39dca73556d19223ac0f7b0dd279be1917d42fa9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [39dca73556d19223ac0f7b0dd279be1917d42fa9] - Code Rule Check OK.

**Comment by junmin-kim** [comment]
@jsdosa Would you review this PR?

**Comment by seinfra** [comment]
Target : [2472a80aefb9f3ba008ae879754e20f850fcfcb1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2472a80aefb9f3ba008ae879754e20f850fcfcb1] - Code Rule Check OK.

File: framework/src/media/MediaPlayerImpl.cpp [hunk 1]
```diff
@@ -56,9 +68,11 @@ player_result_t MediaPlayerImpl::create()
 
 void MediaPlayerImpl::createPlayer(player_result_t &ret)
 {
-	medvdbg("MediaPlayer Worker : createPlayer\n");
+	LOG_STATE_INFO(mCurState);
+
 	if (mCurState != PLAYER_STATE_NONE) {
-		meddbg("MediaPlayer create fail : wrong state\n");
+		meddbg("%s Fail : invalid state\n", __func__);
+		LOG_STATE_DEBUG(mCurState);
```
Comment (JeongJunSik) [inline framework/src/media/MediaPlayerImpl.cpp]: in this case, 
line 71 : LOG_STATE_INFO(mCurState); 
and
line 75 : LOG_STATE_DEBUG(mCurState);
there are the same result.

i think doesn't need this debug message.
how about think?
or how about setup for info or debug in kconfig.

File: framework/src/media/MediaPlayerImpl.cpp [hunk 2]
```diff
@@ -25,6 +25,18 @@
 #include "audio/audio_manager.h"
 
 namespace media {
+const char *player_state_names[] = {
```
Comment (JeongJunSik) [inline framework/src/media/MediaPlayerImpl.cpp]: I think placing it in the same location as the enum define would reduce the programmer's mistake.
or how about switch syntax.

Comment (junmin-kim) [reply framework/src/media/MediaPlayerImpl.cpp]: If I moved `player_state_names[]` to `MediaPlayerImpl.h`
Below build error occurs.
```
/home/tinyara/github/fork/os/../build/output/libraries/libframework.a(InputDataSource.o):(.data._ZN5media18player_state_namesE+0x0): multiple definition of `media::player_state_names'
/home/tinyara/github/fork/os/../build/output/libraries/libframework.a(MediaPlayer.o):(.data._ZN5media18player_state_namesE+0x0): first defined here
/home/tinyara/github/fork/os/../build/output/libraries/libframework.a(MediaPlayerImpl.o):(.data._ZN5media18player_state_namesE+0x0): multiple definition of `media::player_state_names'
/home/tinyara/github/fork/os/../build/output/libraries/libframework.a(MediaPlayer.o):/home/tinyara/github/fork/framework/src/media/MediaPlayer.cpp:24: first defined here
/home/tinyara/github/fork/os/../build/output/libraries/libframework.a(PlayerWorker.o):(.data._ZN5media18player_state_namesE+0x0): multiple definition of `media::player_state_names'
/home/tinyara/github/fork/os/../build/output/libraries/libframework.a(MediaPlayer.o):/home/tinyara/github/fork/framework/src/media/MediaPlayer.cpp:24: first defined here
```

Comment (JeongJunSik) [reply framework/src/media/MediaPlayerImpl.cpp]: 'const char* const player_state_names'
Then the error should disappear.

**Comment by seinfra** [comment]
Target : [bf2322d5da2d6d7ffb05a20f36f184fa736b46ea] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [bf2322d5da2d6d7ffb05a20f36f184fa736b46ea] - Code Rule Check Result:
framework/src/media/MediaPlayerImpl.h:76: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/MediaPlayerImpl.h:76: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:76: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:76: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:76: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxE)
framework/src/media/MediaPlayerImpl.h:93: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:93: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:93: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:93: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:93: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:93: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:94: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:94: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:94: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:94: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:113: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:118: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:118: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:122: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:123: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:123: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:123: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaPlayerImpl.h:124: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:124: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaPlayerImpl.h:124: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaPlayerImpl.h:124: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaPlayerImpl.h:124: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:80: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/MediaRecorderImpl.h:80: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:80: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxE)
framework/src/media/MediaRecorderImpl.h:83: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:97: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:97: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:97: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:97: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:97: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:97: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:99: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:99: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:99: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:99: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:106: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:107: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:108: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:109: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:113: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:114: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:115: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:115: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:115: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:115: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:116: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:116: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:116: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:116: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:116: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:116: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:116: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:117: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:120: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:120: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:120: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:121: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:121: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:122: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:122: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:122: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:124: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:128: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Comment by Taejun-Kwon** [comment]
Very good.

**Pull request merged**